allennlp_server/commands/server_simple.py

```diff
@@ -142,81 +162,79 @@ def _get_predictor(args: argparse.Namespace) -> Predictor:
-        "--overrides",
-        type=str,
-        default="",
-        help="a JSON structure used to override the experiment configuration",
-    )
-    parser.add_argument(
-        "--static-dir", type=str, help="serve index.html from this directory"
-    )
-    parser.add_argument(
-        "--title",
-        type=str,
-        help="change the default page title",
-        default="AllenNLP Demo",
-    )
-    parser.add_argument(
-        "--field-name",
-        type=str,
-        action="append",
-        help="field names to include in the demo",
-    )
-    parser.add_argument(
-        "--port", type=int, default=8000, help="port to serve the demo on"
-    )
+        subparser.add_argument("--predictor", type=str, required=True, help="name of predictor")
 
-    parser.add_argument(
-        "--include-package",
-        type=str,
-        action="append",
-        default=[],
-        help="additional packages to include",
-    )
+        subparser.add_argument(
+            "--weights-file", type=str, help="a path that overrides which weights file to use",
+        )
 
-    args = parser.parse_args(args)
+        subparser.add_argument(
+            "--cuda-device", type=int, default=-1, help="id of GPU to use (if any)"
+        )
 
-    # Load modules
-    for package_name in args.include_package:
-        import_submodules(package_name)
+        subparser.add_argument(
+            "-o",
+            "--overrides",
+            type=str,
+            default="",
+            help="a JSON structure used to override the experiment configuration",
+        )
 
-    predictor = _get_predictor(args)
+        subparser.add_argument(
+            "--static-dir", type=str, help="serve index.html from this directory"
+        )
+
+        subparser.add_argument(
+            "--title", type=str, help="change the default page title", default="AllenNLP Demo",
+        )
+
+        subparser.add_argument(
+            "--field-name",
+            type=str,
+            action="append",
+            dest="field_names",
+            help="field names to include in the demo",
+        )
+
+        subparser.add_argument(
+            "-h", "--host", type=str, default="127.0.0.1", help="interface to serve the wizard on",
+        )
+
+        subparser.add_argument(
+            "-p", "--port", type=int, default=8000, help="port to serve the demo on"
+        )
 
-    field_names = args.field_name
+        subparser.set_defaults(func=serve)
+
+        return subparser
+
+
+def serve(args: argparse.Namespace) -> None:
+    predictor = _get_predictor(args)
 
     app = make_app(
         predictor=predictor,
-        field_names=field_names,
+        field_names=args.field_names,
         static_dir=args.static_dir,
         title=args.title,
     )
     CORS(app)
 
-    http_server = WSGIServer(("0.0.0.0", args.port), app)
-    print(f"Model loaded, serving demo on port {args.port}")
+    http_server = WSGIServer((args.host, args.port), app)
```

Comment:
I went from 0.0.0.0 to 127.0.0.1 as default for security reasons. If someone leaves it open with 0.0.0.0 when debugging, by accident it's exposing it potentially to the whole internet. I used an option for this so it can be changed back to 0.0.0.0 if needed.

On your side, if you folks are serving this repo, remember adding this option cause it breaks compatibility.

Reply:
That's fine, thanks.